tpm/algorithm/algorithm.go

```diff
@@ -0,0 +1,98 @@
+package algorithm
+
+import (
+	"encoding/json"
+	"slices"
+
+	"github.com/google/go-tpm/legacy/tpm2"
+)
+
+var (
+	algs map[tpm2.Algorithm]string
+)
+
+type Algorithm tpm2.Algorithm
+
+func (a Algorithm) String() string {
+	return algs[tpm2.Algorithm(int(a))]
+}
+
+func (a Algorithm) MarshalJSON() ([]byte, error) {
+	return json.Marshal(a.String())
+}
+
+type AlgorithmSlice struct {
+	algs []Algorithm
+}
+
+func (s *AlgorithmSlice) Supports(algs ...Algorithm) bool {
+	if len(algs) == 0 {
+		return false
+	}
+
+	for _, alg := range algs {
+		if !slices.Contains(s.algs, alg) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func init() {
+	// https://trustedcomputinggroup.org/wp-content/uploads/TCG_TPM2_r1p59_Part2_Structures_pub.pdf
+	algs = map[tpm2.Algorithm]string{
+		// object types
+		tpm2.AlgRSA: "RSA",
+		tpm2.AlgECC: "ECC",
+
+		// encryption algs
+		tpm2.AlgRSAES: "RSAES",
+
+		// block ciphers
+		0x0003:            "3DES",
```

Comment:
All that are not defined in `tpm2` can be made into a `tpm2.Algorithm` like this:
```suggestion
		tpm2.Algorithm(0x0003):            "3DES",
```

Reply:
657022a